### ISSUE TITLE
Teachers page

### ISSUE THREAD

**Issue by sagar-barapatre** [Namanl2001/MERN-Gurujii-dev#2]
Maybe what i am thinking is if we can build a page in the website (button will be besides the google sign in button) where we can call the teachers who are willing to teach. 
for example User is already signed-in, then If the user wants to teach instead of studying then we'll open a portal for them as in where they can register themselves for teaching for that particular pincode where he wants to teach. So that many teachers will be able to join that community and learning a fun process.

Again let me know if you find this interesting..

**Comment by Rimpi246** [comment]
And then if he is  a  user(as teacher) then we can render his bio in the home page so that new teachers get more exposure.
 I would like to work on this issue as a part of GSSOC'21.PLease `/assign` this to me.

**Comment by tanwi1234** [comment]
We can make a UI of home page where there will be two cards one for student and other for teachers. student should click on to show the page of mentors where all the details of every mentor would be display each card for particular teacher and teacher would make their profile by clicking on another be a mentor button, a registration form will display and from where all data will be store in database and will display on mentor pages that can be shown by student and they can easily reach to them
One more option of teacher login button where they should edit or delete their profile whatever they want.

I would like to  work on this as a part of GSSOC'21 .Please assign me.

**Comment by Namanl2001** [comment]
Thanks @tanwi1234 for such a great suggestion i really liked it. 
After clicking on the students page also we can ask for signin and if the user doesn't signin we would not show them the contact details of the teacher profiles. WDYT?
But this would need a great co ordination of the team as some contributors are already working on various other parts of UI enhancement.

**Comment by tanwi1234** [comment]
> Thanks @tanwi1234 for such a great suggestion i really liked it.
> After clicking on the students page also we can ask for signin and if the user doesn't signin we would not show them the contact details of the teacher profiles. WDYT?
> But this would need a great co ordination of the team as some contributors are already working on various other parts of UI enhancement.

Yes, You are right we can ask for signin but I do not opt this one because I was thinking by keeping in mind for small age kids belonging from small town or rural areas who have no idea about all this how to login or sign and due to this they are not able to easily  get the benefits of our website. So,if we can keep this website simple for that small kids it would be much better for them.So, if there will be signin option online for teachers and then a dashboard for teachers appear where they can edit add or delete their profile and all the kids get its benefit by filtering the teachers in their particular area would be much better according to me.

**Comment by tanwi1234** [comment]
I would really want to contribute in this and i will be very happy to work with all team and explore and learn new things.

**Comment by tanwi1234** [comment]
please add labels with gssoc'21

**Comment by Namanl2001** [comment]
Hey @tanwi1234 it would be nice if you can open a discussion in the discussion section of this repo so that other contributors working on the UI part could have same imagination. To avoid duplicate efforts. Thanks

**Comment by Namanl2001** [comment]
@tanwi1234 no response from your side since many days. are you working on this?

**Comment by tanwi1234** [comment]
> @tanwi1234 no response from your side since many days. are you working on this?

Yes,I am working on this. I had made the teachers display page and teacher login and dashboard page where they can edit or add their profile .But I have been trying to setup this project on my local machine but it shows some error so i had made this pages yet separately on my pc. I have also not experienced in react js so it's taking some more time. But i am trying to resolve this error asap.

**Comment by tanwi1234** [comment]
![Screenshot (1091)](https://user-images.githubusercontent.com/65597267/112434341-780afa80-8d69-11eb-90f9-54b2e2c9e6a5.png)

Not able to see teachers page after creating profile

**Comment by Namanl2001** [comment]
> ![Screenshot (1091)](https://user-images.githubusercontent.com/65597267/112434341-780afa80-8d69-11eb-90f9-54b2e2c9e6a5.png)
> 
> Not able to see teachers page after creating profile
> 

cd backend
yarn install
yarn start
By this if the backend server starts then it means everything is fine, then just run yarn dev from root of this repo.
If any err in console while starting the backend server you can comment down the screenshot of the console.

**Comment by tanwi1234** [comment]
Is it necessary to make a page using reactJS like I have created a pages pages using html ,css ,php where there is a login page and there is two option of signin and signup for teachers and then after that a admin dashboard page would appear where they can add or edit there profile...If it is necessary to made frontend in reactJs then i will convert this in jsx file?

**Comment by Namanl2001** [comment]
Yes, It should be in reactJS only.

**Comment by Namanl2001** [comment]
@tanwi1234 are you working on this issue?
